services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/validation/ApplicationValidator.kt

```diff
@@ -2,34 +2,67 @@ package fi.hel.haitaton.hanke.validation
 
 import fi.hel.haitaton.hanke.application.Application
 import fi.hel.haitaton.hanke.application.CableReportApplicationData
+import fi.hel.haitaton.hanke.application.Customer
+import fi.hel.haitaton.hanke.application.CustomerWithContacts
 import fi.hel.haitaton.hanke.isValidBusinessId
+import java.time.ZonedDateTime
 import javax.validation.ConstraintValidator
 import javax.validation.ConstraintValidatorContext
+import mu.KotlinLogging
+
+private val logger = KotlinLogging.logger {}
 
 class ApplicationValidator : ConstraintValidator<ValidApplication, Application> {
-    override fun isValid(application: Application?, context: ConstraintValidatorContext?): Boolean {
-
-        val applicationData =
-            application?.applicationData
-                ?: throw InvalidApplicationDataException("Application data missing.")
-
-        return when (applicationData) {
-            is CableReportApplicationData ->
-                listOfNotNull(
-                        applicationData.customerWithContacts.customer,
-                        applicationData.contractorWithContacts.customer,
-                        applicationData.representativeWithContacts?.customer,
-                        applicationData.propertyDeveloperWithContacts?.customer,
-                    )
-                    .all {
-                        when {
-                            it.registryKey == null -> true
-                            it.registryKey.isValidBusinessId() -> true
-                            else -> throw InvalidApplicationDataException("Y-tunnus invalid.")
-                        }
-                    }
+    override fun isValid(application: Application, context: ConstraintValidatorContext?): Boolean {
+
+        return when (application.applicationData) {
+            is CableReportApplicationData -> validOrThrow(application.applicationData)
+        }
+    }
+
+    private fun validOrThrow(cableReportData: CableReportApplicationData): Boolean {
+        if (!isValidCableReport(cableReportData)) {
+            throw InvalidApplicationDataException("Received cable report contains invalid data.")
         }
+        return true
     }
+
+    private fun isValidCableReport(cableReportData: CableReportApplicationData): Boolean =
+        listOf(
+                validStartAndEnd(cableReportData.startTime, cableReportData.endTime),
+                validCustomersWithContacts(cableReportData.customersWithContacts())
+            )
+            .all { it }
```

Comment:
It would make sense to me to have a list of lambdas, so we don't have to do every validation if the first one fails.
```suggestion
        listOf(
                { validStartAndEnd(cableReportData.startTime, cableReportData.endTime) },
                { validCustomersWithContacts(cableReportData.customersWithContacts()) },
            )
            .all { it() }
```

Reply:
Continuing the validation has some advantages too as it tells what issues (instead of an issue) are present (logs).

But it would not change the outcome, failed validation is a failed validation.

Fixed as suggested.